### PR TITLE
Helpers for supporting kando location push and pull using Kopia

### PR DESCRIPTION
**Pull request by pavannd1**
## Change Overview

- Utility function to find the kopia repo object ID of the backed up stream
- Helpers to create kopia snapshot and read a snapshot
- Kando location push and pull functions with kopia

## Pull request type

Please check the type of change your PR introduces:
- [ ] :construction: Work in Progress
- [ ] :rainbow: Refactoring (no functional changes, no api changes)
- [ ] :hamster: Trivial/Minor
- [ ] :bug: Bugfix
- [x] :sunflower: Feature
- [ ] :world_map: Documentation
- [ ] :robot: Test

## Issues

- #XXX

## Test Plan

- [x] :muscle: Manual
- [ ] :zap: Unit test
- [ ] :green_heart: E2E

### CODE REVIEW

File: pkg/kando/location_push.go [hunk 1]
```diff
@@ -72,3 +74,15 @@ func sourceReader(source string) (io.Reader, error) {
...
+// kopiaLocationPush pushes the data from the source using a kopia snapshot
+func kopiaLocationPush(ctx context.Context, path string, source io.Reader) error {
+	snapID, _, err := kopia.Write(ctx, path, source)
+	if err != nil {
+		return errors.Wrap(err, "Failed to push data using kopia")
+	}
+	if err = output.PrintOutput(kopia.BackupIdentifierKey, snapID); err != nil {
+		return err
+	}
+	return nil
```
Comment (PrasadG193) [inline pkg/kando/location_push.go:87]:
```suggestion
	return output.PrintOutput(kopia.BackupIdentifierKey, snapID)
```

**Review by PrasadG193 — APPROVED** [review]

File: pkg/kopia/utils.go [hunk 2]
```diff
@@ -44,6 +50,9 @@ const (
 	// tlsCertificateKey represents the key used to fetch the certificate
 	// from the secret.
 	tlsCertificateKey = "tls.crt"
+
+	// BackupIdentifierKey is the artifact key used for kopia snapshot ID
+	BackupIdentifierKey = "backupID"
```
Comment (PrasadG193) [inline pkg/kopia/utils.go:55]: probably should use `--backup-id` flag

File: pkg/kando/location_pull.go [hunk 3]
```diff
@@ -62,3 +72,8 @@ func targetWriter(target string) (io.Writer, error) {
 func locationPull(ctx context.Context, p *param.Profile, path string, target io.Writer) error {
 	return location.Read(ctx, target, *p, path)
 }
+
+// kopiaLocationPull pulls the data from a kopia snapshot into the given target
+func kopiaLocationPull(ctx context.Context, backupID, path string, target io.Writer) error {
+	return kopia.Read(ctx, path, backupID, target)
```
Comment (PrasadG193) [inline pkg/kando/location_pull.go:78]: Looking at `kopia.Read` def, I think we are passing wrong arguments here
```suggestion
	return kopia.Read(ctx, backupID, path, target)
```

**Comment by PrasadG193** [comment]
I've also observed that `kando location pull` with kopia doesn't work if we pass absolute path as a target. We can fix that in the followups

**Comment by PrasadG193** [comment]
Failing due to lint checks
```
Status: Downloaded newer image for ghcr.io/kanisterio/build:v0.0.12
Running golangci-lint...
pkg/kando/location_pull.go:47:6: `backupIDFlag` is unused (deadcode)
func backupIDFlag(cmd *cobra.Command) string {
     ^
pkg/kando/location_pull.go:77:6: `kopiaLocationPull` is unused (deadcode)
func kopiaLocationPull(ctx context.Context, backupID, path string, target io.Writer) error {
     ^
pkg/kando/location_push.go:79:6: `kopiaLocationPush` is unused (deadcode)
func kopiaLocationPush(ctx context.Context, path string, source io.Reader) error {
     ^
```
We can add exceptions for now

File: pkg/kando/location_pull.go [hunk 4]
```diff
@@ -35,9 +40,14 @@ func newLocationPullCommand() *cobra.Command {
 			return runLocationPull(c, args)
 		},
 	}
+	cmd.Flags().StringP(backupIDFlagName, "b", "", "Pass the backup ID from the location push command (optional)")
 	return cmd
 }
 
+func backupIDFlag(cmd *cobra.Command) string {
```
Comment (pavannd1) [inline pkg/kando/location_pull.go:47]:
```suggestion
// nolint:deadcode
func backupIDFlag(cmd *cobra.Command) string {
```

File: pkg/kando/location_pull.go [hunk 5]
```diff
@@ -62,3 +72,8 @@ func targetWriter(target string) (io.Writer, error) {
 func locationPull(ctx context.Context, p *param.Profile, path string, target io.Writer) error {
 	return location.Read(ctx, target, *p, path)
 }
+
+// kopiaLocationPull pulls the data from a kopia snapshot into the given target
```
Comment (pavannd1) [inline pkg/kando/location_pull.go:76]:
```suggestion
// kopiaLocationPull pulls the data from a kopia snapshot into the given target
// nolint:deadcode
```

File: pkg/kando/location_push.go [hunk 6]
```diff
@@ -72,3 +74,12 @@ func sourceReader(source string) (io.Reader, error) {
 func locationPush(ctx context.Context, p *param.Profile, path string, source io.Reader) error {
 	return location.Write(ctx, source, *p, path)
 }
+
+// kopiaLocationPush pushes the data from the source using a kopia snapshot
```
Comment (pavannd1) [inline pkg/kando/location_push.go:78]:
```suggestion
// kopiaLocationPush pushes the data from the source using a kopia snapshot
// nolint:deadcode
```

File: pkg/kando/location_pull.go [hunk 7]
```diff
@@ -35,9 +40,15 @@ func newLocationPullCommand() *cobra.Command {
 			return runLocationPull(c, args)
 		},
 	}
+	cmd.Flags().StringP(backupIDFlagName, "b", "", "Pass the backup ID from the location push command (optional)")
 	return cmd
 }
 
+// nolint:unused
```
Comment (pavannd1) [inline pkg/kando/location_pull.go:47]:
```suggestion
// nolint:unused,deadcode
```

File: pkg/kando/location_pull.go [hunk 8]
```diff
@@ -62,3 +73,9 @@ func targetWriter(target string) (io.Writer, error) {
 func locationPull(ctx context.Context, p *param.Profile, path string, target io.Writer) error {
 	return location.Read(ctx, target, *p, path)
 }
+
+// kopiaLocationPull pulls the data from a kopia snapshot into the given target
+// nolint:unused
```
Comment (pavannd1) [inline pkg/kando/location_pull.go:78]:
```suggestion
// nolint:unused,deadcode
```

File: pkg/kando/location_push.go [hunk 9]
```diff
@@ -72,3 +74,13 @@ func sourceReader(source string) (io.Reader, error) {
...
+// nolint:unused
```
Comment (pavannd1) [inline pkg/kando/location_push.go:79]:
```suggestion
// nolint:unused,deadcode
```

**Pull request merged**